### PR TITLE
sql: support migrate directory and add implement mysql planner

### PR DESCRIPTION
**Pull request by a8m**
The 2 commits don't have 2 much in common, but I added the mysql plan implementation to give some idea how the usage looks like.

The rest of the drivers will be added in the next diffs.

### CODE REVIEW

**Review by yonidavidson — APPROVED** [review]
Nice work, maybe add it to atlas next week?

**Comment by a8m** [comment]
> Nice work, maybe add it to atlas next week?

Why next week? ;)

**Review by rotemtam — APPROVED** [review]
looks great. some comments

File: sql/migrate/migrate.go [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+	if err != nil {
+		return nil, nil, err
+	}
+	switch {
+	case n == 0 || n >= len(files):
+		return nil, nil, fmt.Errorf("sql/migrate: invalid number for selected files: %d", n)
+	case n > 0:
+		files = files[:n]
+	}
+	// Files are expected to be sort alphabetically.
```
[the remark this answers is not in the document]
Comment (a8m) [reply sql/migrate/migrate.go:390]: 👍

**Pull request merged**